### PR TITLE
Updated build files to build as non-root

### PR DESCRIPTION
**Pull request by warniaha**
## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by zalsader** [comment]
Could you please explain more about how this works?

**Comment by warniaha** [comment]
When logging into one of the SCC hosts , you login as your seagate user id (mine is 942865).  As that user, clone your repos for DAOS, Ceph & s3-tests.  Set a few environment variables, build each one.  The env variables are described on the confluence pages for building.  The command line for building Ceph uses the DAOS_PATH environment variable to set the appropriate folders. see https://seagate-systems.atlassian.net/wiki/spaces/PUB/pages/882638878/DAOS+RGW+DGW+HOWTO

The changes tell cmake where to look for the files

File: cmake/modules/FindDAOS.cmake [hunk 1]
```diff
@@ -32,6 +32,9 @@ set(DAOS_INCLUDE_DIRS ${DAOS_INCLUDE_DIR} ${DAOS_FS_INCLUDE_DIR})
 set(DAOS_LIBRARIES ${DAOS_LIBRARY} ${DAOS_FS_LIBRARY} ${DAOS_UNS_LIBRARY})
 
 include(FindPackageHandleStandardArgs)
+include_directories( ${PC_DAOS_INCLUDEDIR} )
+link_directories( ${PC_DAOS_LIBDIR} )
+
```
Comment (zalsader) [inline cmake/modules/FindDAOS.cmake:37]: Who fills these variables? What do they contain?

File: src/rgw/CMakeLists.txt [hunk 2]
```diff
@@ -273,6 +273,8 @@ endif()
 if(WITH_RADOSGW_DAOS)
   find_package(DAOS REQUIRED)
   target_link_libraries(rgw_common PRIVATE daos dfs uuid duns)
+  target_include_directories(rgw_common PRIVATE ${PC_DAOS_INCLUDEDIR} )
+  link_directories( ${PC_DAOS_LIBRARY_DIRS} )
```
Comment (zalsader) [inline src/rgw/CMakeLists.txt:277]: Do we need both of these or just the first one?

File: cmake/modules/FindDAOS.cmake (hunk 1 above)
Comment (warniaha) [reply cmake/modules/FindDAOS.cmake:37]: They can be filled in on the command line to build ceph, but are not required.  This change helps cmake find the daos.h header file and the dfs libs.

File: src/rgw/CMakeLists.txt (hunk 2 above)
Comment (warniaha) [reply src/rgw/CMakeLists.txt:277]: Both changes are required.  The first helps to find the files it needs, the second one uses the found files in the build

Comment (warniaha) [reply src/rgw/CMakeLists.txt:277]: One thing to note:  When building this way, the release build will need to modify the rpath of the shared libs in the resulting binary to point to the expected loading path.  Again, building this way is optional, but it does fix the DAOS build issue when assigning the output location of the binaries.  I'm open to other solutions...

**Review by zalsader — APPROVED** [review]

**Pull request merged**